delphi-frontend/src/main/java/au/com/integradev/delphi/symbol/scope/DelphiScopeImpl.java

```diff
@@ -170,6 +170,14 @@ private boolean isAcceptableDuplicate(NameDeclaration declaration) {
 
     Set<NameDeclaration> duplicates = declarationsByName.get(declaration.getImage());
 
+    // Unit imports can clash with other declarations, except other imports
+    if (declaration instanceof UnitImportNameDeclaration) {
+      return duplicates.stream().noneMatch(UnitImportNameDeclaration.class::isInstance);
```

Comment:
I may be misreading your comment - but shouldn't this be `allMatch` if duplicate `UnitImportNameDeclaration`s are allowed?

Reply:
Ah, my comment is ambiguously worded. I mean to say "Unit imports can coexist with other declarations, except other imports". The code is correct.

Reply:
Reworded.